src/go/rpk/pkg/cli/cmd/wasm/deploy.go

```diff
@@ -0,0 +1,106 @@
+package transform
+
+import (
+	"encoding/binary"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"vectorized/pkg/utils"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func NewDeployCommand(
+	fs afero.Fs, createProduce func(bool, int32) (sarama.SyncProducer, error),
+) *cobra.Command {
+	var (
+		description	string
+		jvmPartitioner	bool
+		partition	int32
+	)
+
+	command := &cobra.Command{
+		Use:	"deploy <path>",
+		Short:	"deploy inline WASM function",
+		Args:	cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			path := args[0]
+			fullFileName := filepath.Base(path)
+			fileExt := filepath.Ext(fullFileName)
+			// validate file extension, just allow js extension
+			if fileExt != ".js" {
+				return fmt.Errorf("deploy just allow js extension file")
+			}
+			fileName := strings.Replace(fullFileName, fileExt, "", 1)
+			fileContent, err := utils.ReadFileLinesBinary(fs, path)
+			if err != nil {
+				return err
+			}
+			// create producer
+			producer, err := createProduce(jvmPartitioner, partition)
+
+			if err != nil {
+				return err
+			}
+			return deploy(
+				fileName,
+				fileContent,
+				description,
+				producer,
+			)
+		},
+	}
+
+	command.Flags().StringVar(
+		&description,
+		"description",
+		"",
+		"wasm definition description",
+	)
+
+	command.Flags().BoolVarP(
+		&jvmPartitioner,
+		"jvm-partitioner",
+		"j",
+		false,
+		"Use a JVM-compatible partitioner. If --partition is passed"+
+			" with a positive value, this will be overridden and"+
+			" a manual partitioner will be used.",
+	)
+
+	command.Flags().Int32VarP(
```

Comment:
I don't think the user should be able to change the partition that the message is produced to.